src/scripting/StelMainScriptAPI.cpp

```diff
@@ -342,8 +342,29 @@ QStringList StelMainScriptAPI::getAllTimezoneNames()
 
 void StelMainScriptAPI::screenshot(const QString& prefix, bool invert, const QString& dir, const bool overwrite, const QString &format)
 {
+	const bool allowUnsafe = qApp->property("allow_unsafe").toBool();
+	const bool allowed = allowUnsafe || (
+		// Only allow writing in the default directory:
+		dir.isEmpty()
+		// Avoid generating hidden file names:
+		&& !prefix.isEmpty()
+		&& !format.isEmpty()
+		// Protection against path traversal:
+		&& !prefix.contains("/")  && !format.contains("/")
+		// Protection against path traversal (Windows):
+		&& !format.contains("\\") && !prefix.contains("\\")
+		// Protect against using alternate data streams (Windows)?
+		&& !prefix.contains(":") && !format.contains(":")
+		// Make sure we don't overwrite config.ini and such:
+		&& format != "ini");
```

Comment:
only image formats supported by the current build can be stored. .ini is not a supported image type.